docs/markdown/Machine-files.md

```diff
@@ -8,10 +8,34 @@ environments](Native-environments.md).
 ## Sections
 
 The following sections are allowed:
+- constants
 - binaries
 - paths
 - properties
 
+### constants
+
+*Since 0.55.0*
+
+String and list concatenation is supported using the `+` operator.
+Entries defined in the `[constants]` section can be used in any other section
+after its definition (i.e. should usually be first section), entries in any other
+section can be used only within that same section and only after it has been
+defined.
+
+```ini
+[constants]
+toolchain = '/toolchain'
+common_flags = ['--sysroot=' + toolchain + '/sysroot']
+
+[properties]
+c_args = common_flags + ['-DSOMETHING']
+cpp_args = c_args + ['-DSOMETHING_ELSE']
+
+[binaries]
+c = toolchain + '/gcc'
+```
+
```

Comment:
Another example where I can think to use this (with layering is something like):
```ini
[constants]
arch = 'aarch64-linux-gnu'
```

```ini
[binaries]
c = arch + '-gcc'
cpp = arch + '-g++'
strip = arch + '-strip'
pkgconfig = arch + '-pkg-config'
# etc
```

You don't necessarily need to document this, but I'm certain other people will think of the same thing.

Reply:
Yes that's the kind of usage I had in mind too, probably worth mentioning it explicitly in the doc, I'll add.

The other usage I still have in mind is adding `--cross-file-constant toolchain=/path/to/my/toolchain`. The idea is a project could ship their cross file and user would only have to specify from command line where they installed the SDK toolchain without having to write extra file. That's in a separate PR that will need to be adapted to now syntax proposed here: https://github.com/mesonbuild/meson/pull/5859.

Reply:
Mentioned composition in the doc and modified the unit test to verify it works.